php/libraries/LorisForm.class.inc

```diff
@@ -693,10 +693,10 @@ class LorisForm
                 }
             }
         }
-        // Always sanitize user-controlled input
-        if (!is_array($newValue)) {
-            $newValue = htmlspecialchars($newValue);
-        }
+        //        // Always sanitize user-controlled input
```

Comment:
I don't like this fix because it removes an important security measure. I understand though that the double-escaping bug has caused data loss which is a severe issue. 

I'll see if I can come up with an alternate fix this week. If I can't figure it out then I think we'll have to go ahead with this and find another way to address XSS issues.

Reply:
What data loss? Double escaped entries can be restored with a script. It's very unlikely that a patient entered something like `&amp;` on a form.

Reply:
due to the number of `&amp;` added several fields got truncated by mySQL

Reply:
@driusan This was discussed in a group chat on Slack and you can view the details there. `<`, `>`, and `&` seem to be common characters used by clinicians when entering instruments by hand (!) and so there is a need to transcribe them literally into forms.

Since they can't be black-listed, we need to do better at displaying these characters to the front-end.

Reply:
@ridz1208 The problem arose because of MySQL 5.1 though right? If CCNA can't/won't upgrade then this might be a project override rather than something that needs to go into core.

Reply:
it's mostly the `"` that are used by clinicians.

and @johnsaigle not sure I see what can be an override. everyone still has the issue, it's just that the other people's data wont get truncated if they are on more recent versions of mysql, they will still have the double escaping issue. and given that this bug existed since LORIS 17, I think we should cover a wide range of versions in our fix